src/rsz/src/RepairSetup.cc

```diff
@@ -569,6 +601,134 @@ bool RepairSetup::swapPins(PathRef* drvr_path,
   return false;
 }
 
+// Remove driver if
+// 1) it is a buffer without attributes like dont-touch
+// 2) it doesn't create new max fanout violations
+// 3) it doesn't create new max cap violations
+// 4) it doesn't worsen slack
+bool RepairSetup::removeDrvr(PathRef* drvr_path,
+                             LibertyCell* drvr_cell,
+                             int drvr_index,
+                             PathExpanded* expanded)
+{
+  // TODO:
+  // 1. add setup slack check (based on Elmore delay?)
+  // 2. check dont-touch on nets
+  // 3. add slew check
+  if (drvr_cell && drvr_cell->isBuffer()) {
+    Pin* drvr_pin = drvr_path->pin(this);
+    Instance* drvr = network_->instance(drvr_pin);
+
+    // Don't remove buffers from previous sizing, pin swapping, rebuffering, or
+    // cloning because such removal may lead to an inifinte loop or long runtime
+    std::string reason;
+    if (resizer_->all_sized_inst_set_.count(drvr)) {
+      reason = "it has been resized";
+    }
+    if (resizer_->all_swapped_pin_inst_set_.count(drvr)) {
+      reason = "its pins have been swapped";
+    }
+    if (resizer_->all_inserted_buffer_set_.count(drvr)) {
+      reason = "it was from rebuffering";
+    }
+    if (resizer_->all_cloned_inst_set_.count(drvr)) {
+      reason = "it has been cloned";
+    }
```

Comment:
```suggestion
    else if (resizer_->all_swapped_pin_inst_set_.count(drvr)) {
      reason = "its pins have been swapped";
    }
    else if (resizer_->all_inserted_buffer_set_.count(drvr)) {
      reason = "it was from rebuffering";
    }
    else if (resizer_->all_cloned_inst_set_.count(drvr)) {
      reason = "it has been cloned";
    }
```
I'm assuming that only one reason can happen?

Reply:
No.   There may be multiple overlapping reasons, and the last reason is the ultimate one.  But this is for debugging only, so this really doesn't matter much.

Reply:
In that case you could reverse the order and then do else if. Or the compiler might be smart enough to optimize it, who knows.